tests/integration/sharding_tests/test_sharding_relations.py

```diff
@@ -50,16 +61,209 @@ async def test_build_and_deploy(ops_test: OpsTest) -> None:
         application_name=SHARD_ONE_APP_NAME,
     )
 
+    # Will be enabled after DPE-5040 is done
+    # await ops_test.model.deploy(
+    #     MONGOS_APP_NAME,
+    #     channel="6/edge",
+    # )
+
     await ops_test.model.deploy(S3_APP_NAME, channel="edge")
 
     await ops_test.model.wait_for_idle(
         apps=[
+            APP_CHARM_NAME,
+            S3_APP_NAME,
+            REPLICATION_APP_NAME,
             CONFIG_SERVER_ONE_APP_NAME,
             CONFIG_SERVER_TWO_APP_NAME,
             SHARD_ONE_APP_NAME,
         ],
         idle_period=20,
         raise_on_blocked=False,
+        raise_on_error=False,  # TODO: remove raise_on_error when we move to juju 3.5 (DPE-4996)
+    )
+
+
+@pytest.mark.group(1)
+@pytest.mark.abort_on_fail
+async def test_only_one_config_server_relation(ops_test: OpsTest) -> None:
+    """Verify that a shard can only be related to one config server."""
+    await ops_test.model.integrate(
+        f"{SHARD_ONE_APP_NAME}:{SHARD_REL_NAME}",
+        f"{CONFIG_SERVER_ONE_APP_NAME}:{CONFIG_SERVER_REL_NAME}",
+    )
+
+    with pytest.raises(JujuAPIError) as juju_error:
```

Comment:
Could you explain how is this error raised to be capture here? in the context of a hook?